File: client/skills/skills.js
const Ids = require('../public/src/common/model/skills').Ids;
const Types = require('../public/src/common/model/skills').Types;
const Targets = require('../public/src/common/model/skills').Targets;


module.exports.skills = {
    [Ids.PUNCH]: {
        id: Ids.PUNCH,
        key: "punch",
        name: "Punch",
        icon: "fist",
        type: Types.ATTACK,
        damage: 5,
        range: 1,
        cooldown: 1000,
        target: Targets.ENEMIES
    },
    [Ids.BOW_SHOT]: {
        id: Ids.BOW_SHOT,
        key: "bow-shot",
        icon: "bow",
        type: Types.ATTACK,
        damage: 10,
        range: 5,
        projectile: 'arrow.png',
        cooldown: 2000,
        target: Targets.ENEMIES
    },
    [Ids.SWORD_HIT]: {
        id: Ids.SWORD_HIT,
        key: "sword-hit",
        name: "Sword Hit",
        icon: "stone_sword",
        type: Types.ATTACK,
        damage: 20,
        range: 1,
        cooldown: 1500,
        target: Targets.ENEMIES
    }
};

File: server/server-core/src/main/java/dzida/server/core/skill/Skills.java
package dzida.server.core.skill;

import dzida.server.core.basic.entity.Id;

import java.util.Map;

public class Skills {
    public static class Types {
        public final static int ATTACK = 0;
        public final static int BUILDING = 1;
        public final static int GATHER = 2;
    }

    public static class Target {
        public final static int ENEMIES = 0;
    }

    public static class Ids {
        public final static Id<Skill> PUNCH = new Id<>(0);
        public final static Id<Skill> BOW_SHOT = new Id<>(1);
        public final static Id<Skill> SWORD_HIT = new Id<>(2);
    }

    public static Map<Id<Skill>, Skill> skills;

    public static Skill get(Id<Skill> skillId) {
        return skills.get(skillId);
    }
}
